File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        double suma=0;
        String number=Integer.toString(numberToCheck);
        String[] num=number.split("");
        int tam=num.length;
        for (String s : num) {
            int valor = Integer.parseInt(s);
            suma = suma + Math.pow(valor, tam);
        }
        return suma == (double) numberToCheck;
    }
}

File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
    local total, str = 0, tostring(number)

    for num in str:gmatch("%d") do
        total = total + (tonumber(num) ^ #str)
    end

    return total == number
end

return ArmstrongNumbers